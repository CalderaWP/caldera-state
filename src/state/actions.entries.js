/**
 * Name of action to set a page of entries
 *
 * @type {string}
 */
export const SET_ENTRIES = 'SET_ENTRIES';

/**
 * Action to set a page of entries
 * @param {String} formId ID of form these entries belong to.
 * @param {Number} pageNumber Page number for these entries
 * @param {Array} entries Collection of entries
 * @return {{type: string, formId: *, pageNumber: *, entries: *}}
 */
export const setEntries = (formId, pageNumber, entries) => {
	return {
		type: SET_ENTRIES,
		formId,
		pageNumber,
		entries
	};
};

