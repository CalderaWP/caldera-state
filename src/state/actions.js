export const SET_FORMS = 'SET_FORMS';
export const SET_FORM = 'SET_FORM';
export const ADD_FORM_PREVIEW = 'ADD_FORM_PREVIEW';
export const NEW_FORM = 'NEW_FORM';

/**
 * Action creator for setting a form in state
 *
 * @param form
 * @returns {{type: string, form: *}}
 */
export const setForm = (form) => {
    return {
        type: SET_FORM,
        form: form
    }
};

/**
 * Action creator for setting all forms in state
 *
 * @param forms
 * @returns {{type: string, forms: *}}
 */
export const setForms = ( forms ) => {
    return {
        type: SET_FORMS,
        forms:forms
    };
};

/**
 * Action creator for setting one form in state
 *
 * @param formId
 * @param preview
 * @returns {{type: string, formId: *, preview: *}}
 */
export const  addFormPreview = (formId,preview)=> {
    return {
        type: ADD_FORM_PREVIEW,
        formId: formId,
        preview:preview
    }
};


/**
 * Action creator to add an empty form to state
 *
 * @returns {{type: string}}
 */
export const newForm = () => {
    return {
        type: NEW_FORM
    }
}