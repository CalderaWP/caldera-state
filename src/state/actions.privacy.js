export const SET_EDIT_FORM = 'SET_EDIT_FORM';
export const UNSET_EDIT_FORM = 'UNSET_EDIT_FORM';
/**
 * Set the current form to edit in privacy settings
 *
 * @param {String} formId
 * @returns {{type: string, formId: *}}
 */
export const setEditForm = (formId ) => {
    return {
        type: SET_EDIT_FORM,
        formId: formId
    }
};

/**
 * Unset the current form to edit in privacy settings
 *
 * @since 1.7.0
 *
 * @returns {{type: string}}
 */
export const unsetEditForm = () => {
    return {
        type: UNSET_EDIT_FORM,
    }
};
