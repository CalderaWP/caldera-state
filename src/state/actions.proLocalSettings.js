/**
 *
 * @type {string}
 */
export const UPDATE_PRO_SETTINGS = 'CALDERA_FORMS/PRO/SETTINGS/UPDATE';

/**
 *
 * @type {string}
 */
export const UPDATE_PRO_FORM_SETTINGS = 'CALDERA_FORMS/PRO/SETTINGS/FORM/UPDATE';

/**
 * Action creator for updating the CF Pro settings
 * @param  {Object} settings
 * @return {{type: string, settings: *}}
 */
export const updateCfProSettings = (settings)  => {
	return {
		type: UPDATE_PRO_SETTINGS,
		settings
	}
};

/**
 * Action creator for updating the CF Pro settings for one form
 * @param {String} formId
 * @param  {Object} settings
 * @return {{type: string, formId: *, settings: *}}
 */
export const updateCfProFormSetting = (formId,settings) => {
	return {
		type: UPDATE_PRO_FORM_SETTINGS,
		formId,
		settings
	}
};