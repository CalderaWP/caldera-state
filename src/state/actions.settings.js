/**
 *
 * @type {string}
 */
export const SETTINGS_STYLE_INCLUDES = 'CALDERA_FORMS/SETTINGS/STYLE_INCLUDES';

/**
 *
 * @type {string}
 */
export const SETTINGS_OTHER = 'CALDERA_FORMS/SETTINGS/OTHER';

/**
 * Action creator for updating style includes settings
 * @param {Object} styleIncludes
 * @return {{type: string, styleIncludes: *}}
 */
export const updateStyleIncludes = (styleIncludes) => {
	return {
		type: SETTINGS_STYLE_INCLUDES,
		styleIncludes
	};

};

/**
 * Action creator for updating other global settings
 *
 * @param {Object} settingsOther
 * @return {{type: string, settingsOther: *}}
 */
export const updateOtherSettings = (settingsOther) => {
	return {
		type: SETTINGS_OTHER,
		settingsOther
	};

};
