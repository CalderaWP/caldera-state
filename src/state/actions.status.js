export const START_SPIN = 'START_SPIN';
export const STOP_SPIN = 'STOP_SPIN';
export const CLOSE_STATUS_INDICATOR = 'CLOSE_STATUS_INDICATOR';
export const UPDATE_STATUS_INDICATOR = 'UPDATE_STATUS_INDICATOR';

/**
 * Dispatch action to start spinner
 * @returns {{type: string}}
 */
export const startSpinner = () => {
	return {
		type: START_SPIN
	}
};

/**
 * Dispatch action to stop spinner
 *
 *
 * @returns {{type: string}}
 */
export const stopSpinner = () => {
	return {
		type: STOP_SPIN
	}
};

/**
 * Dispatch action to close (hide) status indicator
 *
 * @returns {{type: string}}
 */
export const closeStatus = () => {
	return {
		type: CLOSE_STATUS_INDICATOR
	}
};

/**
 * Update the status indicator
 * 
 * @param {String|Boolean} message Message to show in status indicator
 * @param {Boolean} success Optional. If true, the default, background is green for success. If false, red for failure.
 * @param {Boolean} show Optional. If true, the default, status indicator will show
 * @returns {{type: string, message: *, show: boolean, success: boolean}}
 */
export const updateStatus = (message, success = true, show = true) => {
	if( ! message ){
		message = !success ? 'Error' : 'Success';
	}

	return {
		type: UPDATE_STATUS_INDICATOR,
		message,
		show,
		success
	}
};