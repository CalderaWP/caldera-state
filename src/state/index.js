import {
	setForms,
	setForm,
	addFormPreview
} from './actions';

import {
	getForm,
	getForms,
	getFormPreview,
	getFormPreviews
} from './selectors';

import {
	formsReducer,
} from './reducers';


import {
	combineReducers,createStore
} from 'redux';

const actionFunctions = {
	setForms,
	setForm,
	addFormPreview
};

const selectorFunctions = {
	getForm,
	getForms,
	getFormPreview,
	getFormPreviews
};


export const CALDERA_FORMS_STORE_SLUG = 'CALDERA_FORMS/FORMS';
/**
 * Caldera Forms Redux-store
 *
 * @type {{reducer: formsReducer, actions: {setForms: setForms, setForm: setForm, addFormPreview: addFormPreview}, selectors: {getForm: getForm, getForms: getForms, getFormPreview: getFormPreview, getFormPreviews: getFormPreviews}}}
 */
export const calderaFormsFormState = {
	reducer: formsReducer,
	actions: actionFunctions,
	selectors: selectorFunctions,
};

const reducers = {};
reducers[CALDERA_FORMS_STORE_SLUG] = calderaFormsFormState.reducer;

export const calderaFormsReduxStore = createStore(
	combineReducers(reducers),
	window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

