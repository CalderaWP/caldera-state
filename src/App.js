import React from 'react';
import logo from './logo.svg';
import './App.css';
import {FormList} from './FormList';
import { connect } from 'react-redux';
import {CALDERA_FORMS_STORE_SLUG} from './state';
import  {
	getForms,
} from './state/selectors';
import {
	setForms,
	setForm,
	newForm
} from './state/actions';

const App = (props) => {
	return (
		<div className="App">
			<header className="App-header">
				<img src={logo} className="App-logo" alt="logo" />
				<h1 className="App-title">Caldera State Example</h1>
			</header>
			<div>
				<FormList
					forms={props.forms}
					setForms={props.setForms}
					onFormUpdate={(newValue) => {
						props.setForm(newValue);
					}}
				/>
				<button
					onClick={props.addForm}
				>
                Add Form
				</button>
			</div>
		</div>
	);

};

const mapStateToProps = state => {
	state = state[CALDERA_FORMS_STORE_SLUG];
	let appState = {
		forms: getForms(state)

	};
	return appState;
};

const mapDispatchToProps = (dispatch) => {
	return {
		setForms: (forms) => {
			dispatch(setForms(forms));
		},
		setForm:(form) => {
			dispatch(setForm(form));
		},
		addForm(){
			dispatch(newForm());
		}
	};
};



export default connect(
	mapStateToProps,
	mapDispatchToProps
)(App);

