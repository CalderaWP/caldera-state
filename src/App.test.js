import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { Provider } from 'react-redux';

import {calderaFormsReduxStore} from './state';


describe( 'example',  ()=> {
	it('App renders with state without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<Provider store={calderaFormsReduxStore}>
				<App />
			</Provider>, div );
	});
});

