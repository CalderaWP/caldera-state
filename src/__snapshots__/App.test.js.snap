// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example Example component Lists forms 1`] = `
<div>
  <div>
    <div
      id="form-list-cf1"
    >
      <h2>
        Form One
      </h2>
      <input
        onChange={[Function]}
        type="text"
        value="Form One"
      />
    </div>
  </div>
</div>
`;

exports[`example Example component Shows a form 1`] = `
<div
  id="form-list-cf1"
>
  <h2>
    Form One
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value="Form One"
  />
</div>
`;
